go-controller/pkg/ovn/egressip.go

```diff
@@ -425,6 +425,15 @@ func (oc *Controller) setNodeEgressReachable(nodeName string, isReachable bool)
 
 func (oc *Controller) addEgressNode(egressNode *kapi.Node) error {
 	klog.V(5).Infof("Egress node: %s about to be initialized", egressNode.Name)
+	if stdout, stderr, err := util.RunOVNNbctl(
+		"set",
+		"logical_switch_port",
+		types.EXTSwitchToGWRouterPrefix+types.GWRouterPrefix+egressNode.Name,
+		"options:nat-addresses=router",
```

Comment:
According to the docs, this will also cause us to gratuitous ARP "for all load balancer IP addresses defined on the options:router-port’s logical router"... does that mean we'll start ARPing for service IPs? Or possibly external IPs that we aren't supposed to be explicitly claiming?

It might be better to use the explicitly-enumerated form of `options:nat-addresses`.

Reply:
No, our load balancers (used for services) are attached to the logical switch. And when that doc says "external IP" it only references the OVN denominated field "external IP" used for an OVN NAT entry, ex:

```
    nat 8a390cd4-85d7-4228-bfe0-f5ea4ff7fd0e
        external ip: "172.18.0.45"
        logical ip: "10.244.0.4"
        type: "snat"
```

these are only created for egress IP assignments on the GW router across our code base. So I don't expect to start ARPing for more than egress IP assignments. 

That being said, I missed that part in the doc allowing you to specify specific MAC and IP addresses for GARP. I agree that it's probably better to use that, so I'll re-write this and see if it works.

Reply:
Actually, I am not a fan of doing that. It over-complicates the code, and will probably introduce more instability than it will try to solve. The problem is around removing/adding the GARP configuration for specific egress IPs. Given that we will have multiple `EgressIP` objects, I will need to parse the options and correctly sync the IP addresses, which given the API of `ovn-nbctl` is not very straightforward, and while not impossible, I don't think it warrants the change. 

Ex: a typical logical_switch_port with such a GARP configuration might look something like:

```
_uuid               : dde67193-eb32-4048-86b7-27148dc349bf
addresses           : ["02:42:ac:12:00:02"]
dhcpv4_options      : []
dhcpv6_options      : []
dynamic_addresses   : []
enabled             : []
external_ids        : {}
ha_chassis_group    : []
name                : etor-GR_ovn-control-plane
options             : {nat-addresses="02:42:ac:12:00:02 172.18.0.22 172.18.0.25 172.18.0.23 172.18.0.29", router-port=rtoe-GR_ovn-control-plane}
parent_name         : []
port_security       : []
tag                 : []
tag_request         : []
type                : router
up                  : true
```

Which makes removing `172.18.0.25` not very straightforward. 

I prefer keeping it as is, given that it _should_ only GARP for the egress IP addresses defined, and not other IPs.

Reply:
ok, agree that using the simpler version makes sense if it's not going to cause us to GARP for any wrong IPs